### PR TITLE
[#796 ] Add shellcheck lint

### PR DESCRIPTION
**Pull request by pascal-airy**
We add shellcheck as another lint step and run it for all files shell files in infra and scripts folder otherwise we get errors for scripts that we did not write.
I also removed the smoke test script since it wasn't working see #741 and created a lot of linting errors due to the unused variables. We will pick it up again with #866

### CODE REVIEW

File: infrastructure/images/kafka/scripts/configure-kafka.sh [hunk 1]
```diff
@@ -1,27 +1,28 @@
 #!/bin/bash
 set -eo pipefail
 
-BROKER_ID=$(echo ${POD_NAME} | awk -F "-" '{ print $NF; }')
+BROKER_ID=$(echo "${POD_NAME}" | awk -F "-" '{ print $NF; }')
 BROKER_PORT=$((AIRY_FIRST_LISTENER_PORT + BROKER_ID))
-LISTENERS=$(echo ${AIRY_LISTENERS} | \
-  sed "s/AIRY_BROKER_PORT/${BROKER_PORT}/g")
-ADVERTISED_LISTENERS=$(echo ${AIRY_ADVERTISED_LISTENERS} | \
+LISTENERS="${AIRY_LISTENERS//AIRY_BROKER_PORT/${BROKER_PORT}}"
```
Comment (pascal-airy) [inline infrastructure/images/kafka/scripts/configure-kafka.sh:6]: nice is much nicer

File: scripts/bootstrap.sh [hunk 2]
```diff
@@ -65,13 +65,13 @@ fi
 if ! command -v VBoxManage &> /dev/null
 then
     printf "\nVirtualbox binary not found. Attempting to install...\n"
-    VIRTUALBOX_VERSION="2.2.10"
```
Comment (pascal-airy) [inline scripts/bootstrap.sh:68]: this var was not used at all so I added it to the download

File: scripts/bootstrap.sh [hunk 3]
```diff
@@ -87,13 +87,13 @@ then
             osInfo[/etc/centos-release]="sudo yum -y install"
             osInfo[/etc/fedora-release]="sudo dnf -y install"
             package_manager=""
-            for f in ${!osInfo[@]}
```
Comment (pascal-airy) [inline scripts/bootstrap.sh:90]: the error shellcheck gave me here was a bit weird. I'd like @ljupcovangelski to have a closer look

**Review by lucapette — APPROVED** [review]
👏 👏 👏 

And thank you @bitboxer for suggesting shellcheck!

**Pull request merged**